docs/developer-docs/integrations/exchange-rate/exchange-rate-canister.md

```diff
@@ -0,0 +1,141 @@
+
+- If the request can be served from the cache, the actual cost is 200M cycles.
+- If both assets are fiat currencies, the cost is 200M cycles.
+- If one of the assets is a fiat currency or the cryptocurrency USDT, the cost is 2.6B cycles.
+- If both assets are cryptocurrencies, the cost is 5B cycles.
+
+The remaining cycles are returned to the requesting canister. Note that at least 10M cycles are charged even in case of an error in order to mitigate the risk of a denial-of-service attack.
+
+:::info
+Note: The cycles cost of these requests are expected to be decreased in the next upgrade of the XRC. 
+:::
+
+## Exchange rate canister demo
+
+This sample code illustrates how to interact with the exchange rate canister on the Internet Computer.
+
+### Prerequisites
+
+-   [x] Download and install the IC SDK package as described in the [download and install](/developer-docs/setup/install/index.mdx) page.
+
+-   [x] Download and install [Nodejs and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
+
+-   [x] Download and install [git](https://git-scm.com/downloads).
+
+Then, clone the sample repository, install the dependencies, and build the canisters with the following commands:
+
+```
+git clone git@github.com:THLO/xrc_demo.git
+cd xrc_demo/
+npm install
+dfx canister create xrc_demo
+```
+
+### Deployment
+
+To test the XRC demo locally, first start the local execution environment with the command:
+
+```
+dfx start --background
+```
+
+Then, deploy the canisters with the command:
+
+```
+dfx deploy --with-cycles 10000000000
+```
+
+Note that the `--with-cycles` parameter is required since cycles must be sent with every request to the exchange rate canister. For more information on the cycles cost, review the [cycles cost](#cycles-cost) section above. 
+
+The output of this command will resemble the following:
+
+```
+Installing code for canister xrc, with canister ID br5f7-7uaaa-aaaaa-qaaca-cai
+Installing code for canister xrc_demo, with canister ID be2us-64aaa-aaaaa-qaabq-cai
+Deployed canisters.
+URLs:
+  Backend canister via Candid interface:
+    xrc: http://127.0.0.1:8080/?canisterId=bw4dl-smaaa-aaaaa-qaacq-cai&id=br5f7-7uaaa-aaaaa-qaaca-cai
+    xrc_demo: http://127.0.0.1:8080/?canisterId=bw4dl-smaaa-aaaaa-qaacq-cai&id=be2us-64aaa-aaaaa-qaabq-cai
+```
+
+Open the link for the `xrc_demo` canister in a web browser to view the Candid interface. You will see the following Candid UI:
+
+![Candid UI XRC](../_attachments/Candid-xrc.png)
+
+Alternatively, you can call commands to the `xrc_demo` canister directly from the terminal, such as:
+
+```
+dfx canister call --network ic xrc_demo get_exchange_rate "ICP"
+```
+
+The output will resemble the following, depending asset requested and its current exchange rate. This example uses ICP.
+
+```
+(4.1865 : float64)
+```
+
+To check the current exchange rate for BTC, you can use the command:
```

Comment:
Instead of another command targeting the demo canister, how about calling the exchange rate canister directly?
If your wallet (canister) ID is stored in `WALLET`, you can get the BTC/USD rate as follows:

```
dfx canister call --wallet $WALLET --with-cycles 10000000000 xrc get_exchange_rate '(record { base_asset = record { symbol = "BTC"; class = variant { Cryptocurrency } }; quote_asset = record { symbol = "USD"; class = variant { FiatCurrency } } })' 
```